### ISSUE TITLE
Limit removal for calico-node had no effect because `resources` are not reconciled

### ISSUE THREAD

**Issue by voelzmo** [gardener/gardener-extension-networking-calico#188]
**How to categorize this issue?**

/area networking
/area auto-scaling
/kind bug

**What happened**
The PR https://github.com/gardener/gardener-extension-networking-calico/pull/166 had no effect in our landscapes, because `resources.gardener.cloud/preserve-resources: "true"` is set on the ManagedResources. Therefore, resources (and limits) are not touched when the resource is reconciled and the limit is never removed.

**What you expected to happen**:
https://github.com/gardener/gardener-extension-networking-calico/pull/166 having the desired effect and removing the weird CPU limit of `1000 cores`

**Comment by ScheererJ** [comment]
TLDR: Works as expected.

The annotation is required to prevent gardener resource manager and the calico autoscaler from competing over the resource section overwriting each others changes. However, with the annotation set it is not possible anymore to truly remove the cpu limit. Therefore, we decided to set the cpu limit to a very high (never to be reached) value to have all existing clusters adapt the value. Afterwards the limit was removed with #166.
Now, new clusters will not get cpu limits and old clusters have a limit, which should never come into effect meaning effectively they have no limit.
